nmi_mysql/nmi_mysql.py

```diff
@@ -1,194 +1,122 @@
-
-        except Exception as err:
-            self.logger.error('Failed to connect to db')
-            self.logger.warn('Error:')
-            self.logger.info(err)
-            raise err
 
-        self.logger.info('Connection to mysql')
-        self.connected = True
-        self.handle = con
-        return True
+CONFIG_KEYS = ['host', 'user', 'password', 'db', 'port']
+MAX_POOL_SIZE = 10
 
-    def close(self):
-        try:
-            if self.connected:
-                self.handle.close()
-                self.connected = False
-                self.handle = None
 
-        except Exception as err:
-            self.logger.warn('Failed to close connection')
-            self.logger.warn(err)
-            raise err
-
-        return None
-
-    def query(self, _query, _params=None):
-        """
-            self.handle holds the connection
-            _query is the query
-            _params holds the variables need by the query
-        """
-
-        result = None
-        query = _query
-
-        if _params:
-            query = self.generate_query(_query, _params)
-
-        try:
-            with self.handle.cursor() as cursor:
-                cursor.execute(query, ())
-
-                if query.lower().strip().find('select') == 0:
-                    result = list(cursor.fetchall())
-
-                else:
-                    result = {
-                        'affected_rows': cursor.rowcount
-                    }
-
-        except Exception as err:
-            self.logger.warn(err)
-            raise err
-
-        self.handle.commit()
+class DB(object):
 
-        return result
+    def __init__(self, conf):
+        for c in CONFIG_KEYS:
+            if c not in conf:
+                raise ValueError('Invalid config object')
 
-    def generate_query(self, _query, _params):
+        self.logger = logging.getLogger('database')
+        self.charset = 'utf8mb4'
+        self.con = None
+
+        self.engine = create_engine(
+            self._sql_alchemy_format(conf),
+            pool_size=conf.get('max_pool_size', MAX_POOL_SIZE)
+        )
+
+    def _sql_alchemy_format(self, conf):
+        return ''.join([
+            'mysql+pymysql://',
+            conf['user'] + ':',
+            conf['password'] + '@',
+            conf['host'] + ':',
+            str(conf['port']) + '/',
+            conf['db'],
+            '?charset=' + self.charset
+        ])
+
+    def _generate_query(self, _query, _params):
         query = re.sub('\?', '%s', _query)
 
         if not isinstance(_params, list):
-            return query % self.to_string(_params)
+            return (query % self._to_string(_params), _params)
 
         params = []
-        values = []
+        multi_insert = []
+        other_queries = []
+
         for param in _params:
+            if isinstance(param, list):
+                params.extend(param)
+
+            elif isinstance(param, dict):
```

Comment:
I might be a little sleepy but does this code handle this case?
`db.query('UPDATE users SET %s WHERE name = %s', [{'name': 'ninz'}, 'jasper'])`

Will it not just return `UPDATE users SET ninz WHERE name = jasper`?